packages/node_modules/@webex/plugin-meetings/src/roap/handler.js

```diff
@@ -0,0 +1,166 @@
+/* eslint require-jsdoc:0 */
+/* globals window */
+import {StatelessSparkPlugin} from '@ciscospark/spark-core';
+
+import MeetingCollection from '../meetings/collection';
+import Events from '../events';
+import {ROAP, EVENTS, OFFER} from '../constants';
+import RoapUtil from '../roap/util';
+import RoapCollection from '../roap/collection';
+
+// TODO: set up logging
+const config = window;
+
+const checkForAndHandleErrors = (action, meeting, correlationId) => {
+  if (action && action.type) {
+    if (action.msg && action.msg.messageType && action.msg.errorType) {
+      if (RoapUtil.findError(action.msg.messageType, action.msg.errorType, action.type)) {
+        RoapUtil.handleError(meeting.mediaPeerConnection)
+          .then((res) => {
+            if (res) {
+              RoapCollection.deleteSessionSequence(correlationId, action.msg.seq);
+            }
+          })
+          .catch((err) => {
+            config.console.info('Cannot reset the peer connection', err);
+          });
+        return true;
+      }
+    }
+    if (!RoapUtil.ensureMeeting(meeting, action.type)) {
+      return true;
+    }
+  }
+  return false;
+};
```

Comment:
This method is a little hard to read/follow but I don't have any drastic improvements.